demos/interactive_face_detection_demo/main.cpp

```diff
@@ -7,392 +7,550 @@
-                                       i < ageGenderDetector.maxBatch));
-                if (face->isAgeGenderEnabled()) {
-                    AgeGenderDetection::Result ageGenderResult = ageGenderDetector[i];
-                    face->updateGender(ageGenderResult.maleProb);
-                    face->updateAge(ageGenderResult.age);
-                }
+        cv::Mat ssd_res;
+        out_vector += cv::gout(ssd_res);
 
-                face->emotionsEnable((emotionsDetector.enabled() &&
-                                      i < emotionsDetector.maxBatch));
-                if (face->isEmotionsEnabled()) {
-                    face->updateEmotions(emotionsDetector[i]);
-                }
+        std::vector<cv::Rect> face_hub;
+        out_vector += cv::gout(face_hub);
+
+        std::vector<cv::Mat> out_ages, out_genders;
+        if (!FLAGS_m_ag.empty()) out_vector += cv::gout(out_ages, out_genders);
+
+        std::vector<cv::Mat> out_y_fc, out_p_fc, out_r_fc;
+        if (!FLAGS_m_hp.empty()) out_vector += cv::gout(out_y_fc, out_p_fc, out_r_fc);
+
+        std::vector<cv::Mat> out_emotions;
+        if (!FLAGS_m_em.empty()) out_vector += cv::gout(out_emotions);
 
-                face->headPoseEnable((headPoseDetector.enabled() &&
-                                      i < headPoseDetector.maxBatch));
-                if (face->isHeadPoseEnabled()) {
-                    face->updateHeadPose(headPoseDetector[i]);
+        std::vector<cv::Mat> out_landmarks;
+        if (!FLAGS_m_lm.empty()) out_vector += cv::gout(out_landmarks);
+
+        Visualizer::Ptr visualizer;
+        if (!FLAGS_no_show) {
+            cv::namedWindow("Detection results");
+            visualizer = std::make_shared<Visualizer>();
+            visualizer->enableVisualisations(!FLAGS_m_ag.empty(), !FLAGS_m_em.empty(), !FLAGS_m_hp.empty(), !FLAGS_m_lm.empty());
+        } else {
+            std::cout<< "To close the application, press 'CTRL+C' here" << std::endl;
+        }
+
+        std::list<Face::Ptr> faces;
+        std::ostringstream out;
+        size_t framesCounter = 0;
+        size_t id = 0;
+        cv::VideoWriter videoWriter;
+
+        slog::info << "Start inference " << slog::endl;
+        Timer timer;
+        stream.setSource(cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(FLAGS_i));
+        stream.start();
+
+        while (stream.running())
+        {
+            timer.start("total");
+
+            if (stream.pull(std::move(out_vector))) {
+                if (!FLAGS_no_show && !FLAGS_m_em.empty() && !FLAGS_no_show_emotion_bar) {
+                    visualizer->enableEmotionBar(frame.size(), {"neutral",
+                                                                "happy",
+                                                                "sad",
+                                                                "surprise",
+                                                                "anger"});
```

Comment:
you do it every time, right?

Reply:
Fixed 
Added flag emoBarEnabled
We need this function there because we need to know frame size to enable emotion bar